### PR TITLE
🖨️ Smart contract scaffolding

### PR DESCRIPTION
**Pull request by ccamel**
This PR introduces a basic scaffold mechanism for smart contracts to facilitate the implementation of new smart contracts. This PR doesn't do much apart from copying (and substituting the project name) into the contract directory. But this can easily be improved as required.

Scaffolding uses [ffizer](https://github.com/ffizer/ffizer) for doing the job.

### CODE REVIEW

**Comment by ccamel** [comment]
Following @ErikssonJoakim's review, it seems that installing via the crate demands a Rust version more advanced than what's needed for compiling smart contracts. Additionally, installation with curl leads to binary resolution issues in the `PATH` based on the environment. Therefore, it's best to treat the `ffizer` tool as a necessary prerequisite for executing the task. Developers are responsible for its installation (as detailed in the `README.md`).

**Review by ErikssonJoakim — APPROVED** [review]
Great 😃 Once we've catched up on the rustc version we can implement the install

File: templates/okp4-smart-contract/{{ contract_name }}/Cargo.ffizer.hbs.toml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+okp4-logic-bindings.workspace = true
+okp4-objectarium-client.workspace = true
+okp4-objectarium.workspace = true
+schemars.workspace = true
+serde.workspace = true
+thiserror.workspace = true
+
+[dev-dependencies]
+cw-multi-test.workspace = true
+url = "2.4.0"
```
Comment (bdeneux) [inline templates/okp4-smart-contract/{{ contract_name }}/Cargo.ffizer.hbs.toml:47]:
```suggestion
```
This one too

**Review by bdeneux — CHANGES_REQUESTED** [review]
Nice tool ! 👍 Just a little changes requested to remove the unnecessary dependencies at the beginning of the smart contract development.

**Review by bdeneux — APPROVED** [review]

**Pull request merged**